ledfx/devices/__init__.py

```
@@ -844,3 +804,88 @@
+
+    def is_osc_port_path_seperated(new_device, pre_device):
+        """
+        Check if the new device is osc port and path separated from the pre-existing device
+        """
+        if new_device.type == "osc" and pre_device.type == "osc":
+            if new_device.config["port"] == pre_device.config["port"]:
+                if new_device.config["path"] == pre_device.config["path"]:
+                    if (
+                        new_device.config["starting_addr"]
+                        == pre_device.config["starting_addr"]
+                    ):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)